### PR TITLE
Newsletters: add 291 (2024-02-28)

### PR DESCRIPTION
**Pull request by harding**
- [x] Bitcoin SE section @bitschmidty 
- [x] Lede, releases/RCs, topic entries @harding 
- [x] Topic for ecash @harding 
- [x] Minor update to watchtowers topic @harding

### CODE REVIEW

File: _posts/en/newsletters/2024-02-28-newsletter.md [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+---
+This week's newsletter FIXME
+
+## News
+
+- **Trustless contract for miner feerate futures:** ZmnSCPxj
+  [posted][zmnscpxj futures] to Delving Bitcoin a set of scripts that
+  will allow two parties to conditionally pay each other based on the
+  marginal feerate to include a transaction in a future block.  For
+  example, Alice is a user who expect to include a transaction in block
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-28-newsletter.md:18]:
```suggestion
  example, Alice is a user who expects to include a transaction in block
```

File: _posts/en/newsletters/2024-02-28-newsletter.md [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+  1,000,000 (or a block shortly thereafter).  Bob is a miner who has
+  some chance of mining a block around that time.  They each deposit
+  some of their money into a _funding transaction_ that can be spent one
+  of three ways:
+
+  1. Bob receives back his deposit plus claims Alice's deposit by
+     spending the funding transaction's output in block 1,000,000 (or a
+     block shortly thereafter).  The script they use requires Bob's
+     unilateral spend be a certain minimum size, such as larger than two
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-28-newsletter.md:27]:
```suggestion
     unilateral spend to be a certain minimum size, such as larger than two
```
(or "... requires that Bob's ...")

File: _posts/en/newsletters/2024-02-28-newsletter.md [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+     typical spends.
+
+  2. Alternatively, Alice receive back her deposit plus claims Bob's
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-28-newsletter.md:30]:
```suggestion
  2. Alternatively, Alice receives back her deposit plus claims Bob's
```

File: _posts/en/newsletters/2024-02-28-newsletter.md [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+     deposit spending the funding transaction's output some time after
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-28-newsletter.md:31]:
```suggestion
     deposit spending the funding transaction's output sometime after
```

File: _posts/en/newsletters/2024-02-28-newsletter.md [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+  feerates mean that he doesn't earn as much reward from any blocks he
+  produces.
+
+  If feerates at block 1,000,000 are higher than expected, Bob won't
+  want to include his large spend in a block---it'll cost more in fees
+  than it earns him in profit.  This allows Alice to profit by including
+  her smaller spend in block 1,000,144 (or later).  Profiting at this
+  moment of network-wide high feerates is especially advantageous to
+  Alice as it offsets the high fee cost of including the regular
+  transaction she planned to included in block 1,000,000.
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-28-newsletter.md:53]:
```suggestion
  transaction she planned to include in block 1,000,000.
```

File: _posts/en/newsletters/2024-02-28-newsletter.md [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+  contracts) but also avoids corrupting the counterparty.  For example,
+  if there was a centralized futures market for feerates, Bob and other
+  miners could [accept fees out of band][topic out-of-band fees] or use
+  other tricks to manipulate the apparent feerate; but, with ZmnSCPxj's
+  construction, that's not a risk: Bob's choice about whether or not to
+  use the large-size spend is purely informed by his perspective on
+  current mining and mempool conditions.  That reply also contemplated
+  whether larger miners might have an advantage over smaller miners,
+  with Anthony Towns [providing][towns futures] a payoff table showing
+  that an attempt to game the contract would result is greater profits
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-28-newsletter.md:75]:
```suggestion
  that an attempt to game the contract would result in greater profits
```

File: _posts/en/newsletters/2024-02-28-newsletter.md [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+  for miners using the default transaction selection algorithm.
+
+- **Coin selection for liquidity providers:** Richard Myers
+  [posted][myers cs] to Delving Bitcoin about creating a [coin
+  selection][topic coin selection] algorithm that is optimized for LN
+  nodes offering liquidity via [liquidity advertisements][topic
+  liquidity advertisements].  His post describes an algorithm which he
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-28-newsletter.md:82]:
```suggestion
  liquidity advertisements].  His post describes an algorithm that he
```

File: _posts/en/newsletters/2024-02-28-newsletter.md [hunk 8]
```diff
@@ -0,0 +1,278 @@
...
+    contest period (e.g., the 20 block delay) and creates a re-vaulting
+    transaction with two inputs and one output; the inputs are the
+    trigger transaction's first output and a fee-paying input; the
+    output is a return to the vault.  Because the re-vaulting
+    transaction only has one output but the withdrawal conditions of the
+    script require spending from a trigger transaction with two outputs,
+    Mallory is unable to complete stealing Alice's funds.
+
+    Because the money is returned to the same vault script it
+    started in, Mallory still has the ability to create another trigger
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-28-newsletter.md:140]:
```suggestion
    started in, Mallory can still create another trigger
```

File: _posts/en/newsletters/2024-02-28-newsletter.md [hunk 9]
```diff
@@ -0,0 +1,278 @@
...
+
+  <tr>
+    <th markdown="span">Operational efficiency in best case, i.e. only legitimate spends<br>*(only very roughly estimated by Optech)*</th>
+    <td markdown="span">**2x size of regular single-sig**</td>
+    <td markdown="span">3x size of regular single-sig</td>
+    <td markdown="span">4x size of regular single-sig</td>
+  </tr>
+  </table>
+
+  The prototype had received a small amount of discussion and analysis
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-28-newsletter.md:205]:
```suggestion
  The prototype has received a small amount of discussion and analysis
```

File: _posts/en/newsletters/2024-02-28-newsletter.md [hunk 10]
```diff
@@ -0,0 +1,278 @@
...
+Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
+[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
+Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
+Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
+Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo],
+[Bitcoin Inquisition][bitcoin inquisition repo], and [BINANAs][binana
+repo]._
+
+- [LDK #2770][] begins preparing to later add support for [dual
+  funded channels][topic dual funding].
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-02-28-newsletter.md:260]:
```suggestion
- [LDK #2770][] begins preparing to later add support for
  [dual-funded channels][topic dual funding].
```

File: _posts/en/newsletters/2024-02-28-newsletter.md [hunk 11]
```diff
@@ -0,0 +1,278 @@
...
+
+{% assign day_after_posting = page.date | date: "%s" | plus: 86400 | date: "%Y-%m-%d 15:00" %}
+{% include snippets/recap-ad.md when=day_after_posting %}
+{% include references.md %}
+{% include linkers/issues.md v=2 issues="2770,29422" %}
```
Comment (bitschmidty) [inline _posts/en/newsletters/2024-02-28-newsletter.md:288]:
```suggestion
{% include linkers/issues.md v=2 issues="2770,29442" %}
```

File: _posts/en/newsletters/2024-02-28-newsletter.md [hunk 12]
```diff
@@ -0,0 +1,278 @@
...
+  with Anthony Towns [providing][towns futures] a payoff table showing
+  that an attempt to game the contract would result is greater profits
+  for miners using the default transaction selection algorithm.
+
+- **Coin selection for liquidity providers:** Richard Myers
+  [posted][myers cs] to Delving Bitcoin about creating a [coin
+  selection][topic coin selection] algorithm that is optimized for LN
+  nodes offering liquidity via [liquidity advertisements][topic
+  liquidity advertisements].  His post describes an algorithm which he
+  implemented in a Bitcoin Core [draft PR][bitcoin core #29422].
```
Comment (bitschmidty) [inline _posts/en/newsletters/2024-02-28-newsletter.md:83]:
```suggestion
  implemented in a Bitcoin Core [draft PR][bitcoin core #29442].
```

File: _posts/en/newsletters/2024-02-28-newsletter.md [hunk 13]
```diff
@@ -0,0 +1,303 @@
...
+  [posted][towns lnecash] to Delving Bitcoin about linking
+  "[ecash][topic ecash] mints to the lightning network without losing
+  ecash’s anonymity or adding any additional trust".  His proposal for
+  achieving that goal uses a zero-knowledge contingent payment
+  ([ZKCP][topic acc]) for sending payments to the user of an ecash mint
+  and a process of committing to a hash preimage for withdrawing ecash
+  funds to LN.
+
+  Calle, the lead developer of the [Cashu][] ecash implementation,
+  replied with some concerns but also support for the idea, a reference
```
Comment (bitschmidty) [inline _posts/en/newsletters/2024-02-28-newsletter.md:218]: maybe link replied to https://delvingbitcoin.org/t/ecash-and-lightning-via-zkcp/586/2

File: _posts/en/newsletters/2024-02-28-newsletter.md [hunk 14]
```diff
@@ -232,7 +232,30 @@ answers posted since our last update.*
...
+  User Lagrang3 explains the Transparent Index of Distinct Extended Shares
+  (TIDES) miner payout scheme used by the Ocean mining pool.
+
+- [What data does the Bitcoin Core wallet search for during a blockchain rescan?]({{bse}}121563)
+  Pieter Wuille and Ava Chow summarize how the Bitcoin Core wallet software
+  identifies relevant transactions to a particular legacy or [descriptor][topic descriptors] wallet.
+
+- [How does transaction rebroadcasting for watch-only wallets work?]({{bse}}121899)
+  Ava Chow notes that the transaction rebroadcasting logic is the same
+  regardless of wallet type. However, in order for a watch-only-originated
```
[the remark this answers is not in the document]
Comment (bitschmidty) [reply _posts/en/newsletters/2024-02-28-newsletter.md]: Took this suggestion

File: _posts/en/newsletters/2024-02-28-newsletter.md [hunk 15]
```diff
@@ -0,0 +1,303 @@
...
+{% assign bse = "https://bitcoin.stackexchange.com/a/" %}
+
+- [Why can't nodes have the relay option to disallow certain transaction types?]({{bse}}121734)
+  Ava Chow outlines thoughts about the purpose of [mempool and relay policy][policy series],
+  benefits of more-homogeneous mempools including [fee estimation][topic fee
+  estimation] and [compact block relay][topic compact block relay], and touches
+  on policy workarounds such as miners accepting [fees out-of-band][topic
+  out-of-band fees].
+
+- [What is the Circular Dependency in signing a chain of unconfirmed transactions?]({{bse}}121959)
```
Comment (vostrnad) [inline _posts/en/newsletters/2024-02-28-newsletter.md:242]:
```suggestion
- [What is the circular dependency in signing a chain of unconfirmed transactions?]({{bse}}121959)
```

**Comment by harding** [comment]
Made edits (thanks @LarryRuane @bitschmidty @vostrnad !), added remaining content, and reviewed the SE section (thanks @bitschmidty !).  Thanks everyone!

File: _posts/en/newsletters/2024-02-28-newsletter.md [hunk 16]
```diff
@@ -0,0 +1,309 @@
+---
+title: 'Bitcoin Optech Newsletter #291'
+permalink: /en/newsletters/2024/02/28/
+name: 2024-02-28-newsletter
+slug: 2024-02-28-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter descibes a proposed contract for trustless miner
```
Comment (azuchi) [inline _posts/en/newsletters/2024-02-28-newsletter.md:10]:
```suggestion
This week's newsletter describes a proposed contract for trustless miner
```

File: _posts/en/newsletters/2024-02-28-newsletter.md [hunk 17]
```diff
@@ -0,0 +1,309 @@
...
+feerate futures, links to a coin selection algorithm for LN nodes
+providing dual funding liquidity, details a prototype for a vault using
+`OP_CAT`, and discusses sending and receiving ecash using LN and ZKCPs.
+Also included are our regular sections summarizing popular questions and
+answers from the Bitcoin Stack Exachange, announcing new releases and
+release candidates, and describing recent changes to popular Bitcoin
+infrastructue projects.
```
Comment (azuchi) [inline _posts/en/newsletters/2024-02-28-newsletter.md:17]:
```suggestion
infrastructure projects.
```

File: _posts/en/newsletters/2024-02-28-newsletter.md [hunk 18]
```diff
@@ -0,0 +1,309 @@
...
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter descibes a proposed contract for trustless miner
+feerate futures, links to a coin selection algorithm for LN nodes
+providing dual funding liquidity, details a prototype for a vault using
+`OP_CAT`, and discusses sending and receiving ecash using LN and ZKCPs.
+Also included are our regular sections summarizing popular questions and
+answers from the Bitcoin Stack Exachange, announcing new releases and
```
Comment (azuchi) [inline _posts/en/newsletters/2024-02-28-newsletter.md:15]:
```suggestion
answers from the Bitcoin Stack Exchange, announcing new releases and
```

File: _topics/en/ecash.md [hunk 19]
```diff
@@ -0,0 +1,86 @@
...
+## threshold word count
+#stub: false
+
+## Required.  Use Markdown formatting.  Only one paragraph.  No links allowed.
+## Should be less than 500 characters
+excerpt: >
+  **Ecash** is a type of centralized digital currency that uses blind
+  signatures to prevent the centralized controlling party (the mint) from
+  knowing the balance of any particular user or from learning which
+  users were involved in any transactions.
```
Comment (azuchi) [inline _topics/en/ecash.md:57]:
```suggestion
  users were involved in any transactions.
```

Fix MD009 Trailing spaces error.

**Review by bitschmidty — COMMENTED** [review]
Took suggestions from @azuchi , added a change of my own to the CLN release summary

**Review by bitschmidty — APPROVED** [review]
ACK topics, lede. 

Squashed.

Thank you all!

**Pull request merged**